mingw-w64-openssl/002-relocation.patch

```diff
@@ -37,55 +37,38 @@
              !ENGINE_ctrl_cmd_string(iterator, "DIR_LOAD", "2", 0) ||
 --- a/crypto/x509/x509_def.c
 +++ b/crypto/x509/x509_def.c
-@@ -9,27 +9,44 @@
- 
+@@ -10,6 +10,7 @@
  #include <stdio.h>
+ #include "internal/e_os.h"
  #include "internal/cryptlib.h"
 +#include "pathtools.h"
+ #include "internal/thread_once.h"
  #include <openssl/crypto.h>
  #include <openssl/x509.h>
- 
- const char *X509_get_default_private_dir(void)
+@@ -47,19 +48,19 @@
  {
--    return X509_PRIVATE_DIR;
-+    static char * reloc = NULL;
-+    if (reloc == NULL) {
-+        reloc = single_path_relocation_lib(OPENSSLBIN, X509_PRIVATE_DIR);
-+    }
-+    return reloc;
- }
+     get_windows_default_path(x509_private_dir, "\\private");
+     if (strlen(x509_private_dir) > 0)
+-        x509_private_dirptr = x509_private_dir;
++        x509_private_dirptr = single_path_relocation_lib(OPENSSLBIN, x509_private_dir);
```

Comment:
Previously, the `reloc` variable was a singleton that got initialized once, and then returned all the time. I _think_ that we'd like to have that performance boost back.

(There are more instances of this below.)

Reply:
Ah, I just spotted `DEFINE_RUN_ONCE_STATIC` in the diff, so all is good!